src/main/java/com/bina/varsim/fastqLiftover/FastqLiftOver.java

```diff
@@ -117,37 +130,55 @@ public void doLiftOverDwgsimFastqMap(final MapBlocks mapBlocks, final PrintStrea
         doLiftOverPairedFastq(mapBlocks, new DWGSIMPairedFastqReader(br1, br2, forceFiveBaseEncoding), ps1, ps2);
     }
 
+    public void doLiftOverPbsimFastqMap(final MapBlocks mapBlocks, final PrintStream ps) throws IOException {
+        BufferedReader brFastq = new BufferedReader(new InputStreamReader(decompressStream(fastqFiles.get(0))));
+        BufferedReader brMAF = new BufferedReader(new InputStreamReader(decompressStream(mafFiles.get(0))));
+        BufferedReader brREF = new BufferedReader(new InputStreamReader(decompressStream(refFiles.get(0))));
+
+        doLiftOverPairedFastq(mapBlocks, new PBSIMFastqReader(brREF,brMAF, brFastq, forceFiveBaseEncoding), ps, null);
+    }
+
     public void doLiftOverPairedFastq(final MapBlocks mapBlocks, final PairedFastqReader pfr, final PrintStream ps1, final PrintStream ps2) throws IOException {
         SimulatedReadPair readPair;
         int readCount = 0;
         while ((readPair = pfr.getNextReadPair()) != null) {
+
+            final boolean hasRead2 = readPair.read2 != null;
+            if( hasRead2 != (ps2 != null) ) throw new RuntimeException();
+
             readPair.read1.laneId = laneId;
-            readPair.read2.laneId = laneId;
+            if(hasRead2) {
+                readPair.read2.laneId = laneId;
+            }
 
+            // if read2 not present, set locs2 to output the same as locs1 to preserve pair-ended behavior
             final GenomeLocation loc1 = readPair.read1.locs1.get(0);
-            final GenomeLocation loc2 = readPair.read2.locs2.get(0);
+            final GenomeLocation loc2 = hasRead2 ? readPair.read2.locs2.get(0) : loc1;
             final List<GenomeLocation> newLocs1 = mapBlocks.liftOverInterval(loc1.chromosome, loc1.location, loc1.location + readPair.read1.alignedBases1, loc1.direction);
-            final List<GenomeLocation> newLocs2 = mapBlocks.liftOverInterval(loc2.chromosome, loc2.location, loc2.location + readPair.read2.alignedBases2, loc2.direction);
+            final List<GenomeLocation> newLocs2 = hasRead2 ? mapBlocks.liftOverInterval(loc2.chromosome, loc2.location, loc2.location + readPair.read2.alignedBases2, loc2.direction) : newLocs1;
 
             readPair.read1.locs1 = newLocs1;
             readPair.read1.locs2 = newLocs2;
             readPair.read1.origLocs1 = new ArrayList<>();
             readPair.read1.origLocs2 = new ArrayList<>();
-            readPair.read2.locs1 = newLocs1;
-            readPair.read2.locs2 = newLocs2;
-            readPair.read2.origLocs1 = new ArrayList<>();
-            readPair.read2.origLocs2 = new ArrayList<>();
-
+            ++readCount;
```

Comment:
Since you are writing this to support a pair of reads, this should also reflect that